### PR TITLE
adding http source doc

### PR DESCRIPTION
**Pull request by akihikokuroda**
Part of #522 

Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

### CODE REVIEW

**Review by timflannagan — COMMENTED** [review]
It looks like the code bracket formatting is off here. Can we fix that first, before diving into any of the documentation review here?

**Comment by akihikokuroda** [comment]
@timflannagan Sorry.  I fixed it.

File: docs/sources/http.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+## Authorization
+
+A http source can provide the contents of a compressed archive file with an authorization for the file access by creating a secret in the namespace that the provisioner is deployed.
+Tthe secret is a [Basic authentication secret](https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret)
+and expected to contain `data.username` and `data.password` for the username and password, respectively.
+If the `http.auth.secret.insecureSkipVerify` is set true, the download operation will accept any certificate presented by the server and any host name in that
+certificate. In this mode, TLS is susceptible to machine-in-the-middle attacks unless custom verification is
+used. This should be used only for testing.
```
Comment (tylerslaton) [inline docs/sources/http.md:40]:
```suggestion
An http source can provide authorization for access to private compressed archives by creating a secret in the namespace that the provisioner is deployed.
The secret used in the http source is based around [Basic authentication](https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret)
and is expected to contain `data.username` and `data.password` for the username and password, respectively.
If the `http.auth.secret.insecureSkipVerify` is set true, the download operation will accept any certificate presented by the server and any host name in that
certificate. In this mode, TLS is susceptible to machine-in-the-middle attacks unless custom verification is
used. This should be used only for testing.
```

File: docs/sources/http.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+### Example with authorization
+
+1. Create the secret
+
+```sh
+echo -n 'user name' > username.txt
+echo -n 'password' > password.txt
+kubectl create secret generic gitsecret --type "kubernetes.io/basic-auth" --from-file=username=./username.txt --from-file=password=./password.txt -n rukpak-system
```
Comment (tylerslaton) [inline docs/sources/http.md:49]: For the docs maybe its better to do this in one line without creating two files?

```shell
kubectl create secret generic gitsecret --type "kubernetes.io/basic-auth" --from-literal=username=username --from-literal=password=password 
```

**Review by tylerslaton — COMMENTED** [review]
Had some small nits, but otherwise this looks close to merging.

File: docs/sources/http.md (hunk 1 above)
Comment (akihikokuroda) [reply docs/sources/http.md:40]: I'll update.  Thanks!

File: docs/sources/http.md (hunk 2 above)
Comment (akihikokuroda) [reply docs/sources/http.md:49]: I'll change it.  Thanks!

**Review by tylerslaton — APPROVED** [review]
Awesome, this looks great.

File: docs/sources/http.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+If the `http.auth.secret.insecureSkipVerify` is set true, the download operation will accept any certificate presented by the server and any host name in that
+certificate. In this mode, TLS is susceptible to machine-in-the-middle attacks unless custom verification is
+used. This should be used only for testing.
+
+### Example with authorization
+
+1. Create the secret
+
+```sh
+kubectl create secret generic gitsecret --type "kubernetes.io/basic-auth" --from-literal=username=username --from-literal=password=password -n rukpak-system
```
Comment (joelanford) [inline docs/sources/http.md:47]: Nit: use myusername and mypassword or similar to disambiguate the keys and values

File: docs/sources/http.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      labels:
+        app: my-ahoy
+    spec:
+      provisionerClassName: core-rukpak-io-helm
+      source:
+        http:
+          url: https://github.com/helm/examples/releases/download/hello-world-0.1.0/hello-world-0.1.0.tgz
+          auth:
+            secret:
+              name: gitsecret
```
Comment (joelanford) [inline docs/sources/http.md:71]: Nit: githubsecret?

Not to be confused with a git source secret

**Review by joelanford — APPROVED** [review]
A few minor nits, not looks great! Thanks for working on these!

**Pull request merged**